src/FaviconCache.ts

```diff
@@ -60,21 +74,60 @@ class _FaviconCache {
         return json.data.icon_img;
     }
 
-    private async downloadIndexAndParseFavicon(url: string) {
+    private getBestSizeFromAttribute = (sizesAttr: string): number => {
+        const getSize = (sizeAttr: string) => parseInt(sizeAttr.split(/[xX]/)[0], 10) || 0;
+        const sizes = sizesAttr.split(' ')
+            .map(size => getSize(size))
+            .sort((a, b) => b - a)
+        ;
+        return sizes.length > 0
+            ? sizes[0]
+            : 0
+        ;
+    }
+
+    private findIconsInLinks = (links: Node[]): Icon[] => {
+        const isIcon = (icon: Partial<Icon>): icon is Icon => icon.href != null && icon.size != null;
+        return links.map(link => {
+            const href = this.matchRelAttributes(link, ['shortcut icon', 'icon', 'apple-touch-icon']) || undefined;
+            const sizes = HtmlUtils.getAttribute(link, 'sizes') || '';
+            const size = this.getBestSizeFromAttribute(sizes);
+            return {
+                href,
+                size,
+            };
+        })
+        .filter<Icon>(isIcon);
+    }
+
+    private getBestIcon = (icons: Icon[]): Icon | undefined => {
+        const iconExtensionWeight = (iconHref: string) => iconHref.endsWith('.png')
```

Comment:
Can't we access the image sizes here somehow?
I'm not aware of the reason, but I think there are some cases when a png image uses slightly different extension pn + other character. It's only relevant because I think I've seen this in the case of tumblr or reddit.

Reply:
Theoretically we could, if we downloaded the images and then opened them and queried the dimensions, but I think it's just not worth the effort. This case has been added because I found a website which had both `.ico` and `.png` without any other information and in that case the app should choose png anyhow.

It's possible for the website to provide different sizes of the icons (see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link) and we handle that case with the `getBestSizeFromAttribute` function.

If we found later that some websites use different image extensions than png, we can add them to this function easily later.